com/bl/ttt/tictactoe.py

```diff
@@ -32,27 +32,30 @@ def _is_cell_empty(self, move):
     def make_valid_move(self, move):
         self.cells[move.cell_chosen] = move.player_id
 
-    def is_any_row_complete(self, player_id):
+    def has_won(self, player_id):
+        if self._is_any_row_complete(player_id) or self._is_any_column_complete(player_id) or \
```

Comment:
Ok looking at this again i see you could do this : 
`return self._is_any_row_complete(player_id) or self._is_any_column_complete(player_id) or ...`

Reply:
Sorry to mess you about - I am a bit distracted today !

Reply:
Done.